before_install_linux.sh

```diff
@@ -16,27 +16,38 @@ if [ ! -d /usr/include/freetype ]; then
 fi
 
 if [ -v IMAGEMAGICK_VERSION ]; then
-  wget http://www.imagemagick.org/download/releases/ImageMagick-${IMAGEMAGICK_VERSION}.tar.xz
-  tar -xf ImageMagick-${IMAGEMAGICK_VERSION}.tar.xz
-  cd ImageMagick-${IMAGEMAGICK_VERSION}
+
+  if [ -d ImageMagick-source/ImageMagick-${IMAGEMAGICK_VERSION} ]; then
```

Comment:
Maybe I'm missing something, but it looks like this might use the same cached compilation for the hdri and non-hdri options.

Reply:
Make sense. Sounds this PR is evil. 🙇‍♂️

Reply:
@Watson1978 Haha, well I think it's a good idea. It doesn't make sense for us to compile ImageMagick every time, so it could save us a lot of time. If we could sanitize the configuration options, maybe we could compile the build in a sub-directory based on the options:

```sh
  if [ -d ImageMagick-source/ImageMagick-${IMAGEMAGICK_VERSION}-${CONFIGURE_OPTIONS} ]; then
```

Reply:
@mockdeep That sounds like a good plan because there should really be some changes made to improve the build speed because we are testing so many combinations.

Reply:
Thank you for your advice. I will retry this later. 😄